src/PrestaShopBundle/Translation/View/TranslationsTreeBuilder.php

```diff
@@ -0,0 +1,314 @@
+<?php
+
+/**
+ * 2007-2020 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Translation\View;
+
+use Doctrine\Common\Inflector\Inflector;
+use Symfony\Bundle\FrameworkBundle\Routing\Router;
+
+/**
+ * Builds a domain tree for the translation API
+ */
+class TranslationsTreeBuilder
```

Comment:
We need tests for this class too 😉  it's a critical piece of logic. Building trees is complex, and there are multiple edge-cases that we need to be sure we handle.

However PHPUnit Assertions are not fun for tree testing. Maybe we need to a build a little helper method, that we can use as "we give it one `$translationArray` as input, and we tell it what tree we want at the end, and just run `assertOutputIsAsExpected()` and it tells us yes or no?"

Reply:
I see you have tested this class within another. This is OK 😄  but this piece of php logic is so complex, don't you think it deserves its own set of tests?

Reply:
You're right. I will do a dedicated test for this class